.github/workflows/go-test.yaml

```diff
@@ -14,7 +14,7 @@ jobs:
       - name: Setup Go
         uses: actions/setup-go@v4
         with:
-          go-version: 1.20.x
+          go-version: '^1.20.3'
```

Comment:
can you instead leave this at `1.20.x`
but add the setting:
`check-latest: true`

https://github.com/actions/setup-go#check-latest-version
I believe this will solve our issue